lib/auth/auth_with_roles.go

```diff
@@ -908,6 +986,43 @@ func (a *AuthWithRoles) GenerateUserCerts(ctx context.Context, req proto.UserCer
 		return nil, trace.AccessDenied("this request can be only executed by an admin")
 	}
 
+	if len(req.AccessRequests) > 0 {
```

Comment:
Same as above, I don't think this logic belongs in `AuthWithRoles`. Take a look at `services.Role` and how changes to it are propagated from `APIServer` -> `AuthWithRoles` -> `AuthServer` -> `services.Role`.

Ideally this would be the flow:

1. `APIServer` accepts and authenticates request.
2. `APIServer` validate the input parameters.
3. `AuthWithRoles` performs RBAC checks.
4. `AuthServer` calls `services.AccessRequest` to check/configure the access request.
5. `AuthServer` makes changes to backend.